### PR TITLE
Python Core logger usage in api

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: python/fledge/services/core/api/asset_tracker.py [hunk 1]
```diff
@@ -124,7 +121,7 @@ async def deprecate_asset_track_entry(request: web.Request) -> web.Response:
                             audit_details = {'asset': asset_name, 'service': svc_name, 'event': audit_event_name}
                             await audit.information('ASTDP', audit_details)
                         except:
-                            _logger.warning("Failed to log the audit entry for {} deprecation.".format(asset_name))
+                            _logger.info("Failed to log the audit entry for {} deprecation.".format(asset_name))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/asset_tracker.py:124]: this is warning only, no?

File: python/fledge/services/core/api/auth.py [hunk 2]
```diff
@@ -79,8 +78,8 @@ def __remove_ott_for_user(user_id):
     """Helper function that removes given user_id from OTT_MAP if the user exists in the map."""
     try:
         _user_id = int(user_id)
-    except ValueError as ex:
-        _logger.info("User id given is not an integer.")
+    except ValueError:
+        _logger.error("User id given is not an integer.")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:82]: ID

File: python/fledge/services/core/api/auth.py [hunk 3]
```diff
@@ -323,7 +322,7 @@ async def get_user(request):
             if user_id <= 0:
                 raise ValueError
         except ValueError:
-            _logger.warning("Get user requested with bad user id")
+            _logger.error("Get user requested with bad user id.")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:325]: we can remove this

File: python/fledge/services/core/api/auth.py [hunk 4]
```diff
@@ -323,7 +322,7 @@ async def get_user(request):
...
             raise web.HTTPBadRequest(reason="Bad user id")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:326]: ID

File: python/fledge/services/core/api/auth.py [hunk 5]
```diff
@@ -625,25 +624,24 @@ async def update_password(request):
...
         _logger.warning(msg)
         raise web.HTTPBadRequest(reason=msg)
     except Exception as exc:
         _logger.exception(str(exc))
         raise web.HTTPInternalServerError(reason=str(exc))
 
-    _logger.info("Password has been updated successfully for user id:<{}>".format(int(user_id)))
-
-    return web.json_response({'message': 'Password has been updated successfully for user id:<{}>'.format(int(user_id))})
+    msg = "Password has been updated successfully for user id:<{}>.".format(int(user_id))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:642]: This is a good case of audio logging

Comment (ashish-jabble) [reply python/fledge/services/core/api/auth.py:642]: Audit log info is there and just after the line

https://github.com/fledge-iot/fledge/blob/67dbe4f9d00dfd52f49fcaf1224c40c0d9159387/python/fledge/services/core/api/auth.py#L643

File: python/fledge/services/core/api/browser.py [hunk 6]
```diff
@@ -940,6 +954,7 @@ async def asset_timespan(request):
         raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
     except Exception as exc:
         msg = str(exc)
+        _logger.error("Get timespan of buffered readings for each asset failed. Found error: {}".format(msg))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/browser.py:957]: Failed to get timespan of buffered readings for assets

File: python/fledge/services/core/api/control_service/script_management.py [hunk 7]
```diff
@@ -143,6 +141,7 @@ async def add_schedule_and_configuration(request: web.Request) -> web.Response:
         raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
     except Exception as ex:
         msg = str(ex)
+        _logger.error("Automation script schedule task failed. Found error: {}".format(msg))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/control_service/script_management.py:144]: Task did not fail. 

Failed to add scheduled task for Automation script `<name>`

Comment (praveen-garg) [reply python/fledge/services/core/api/control_service/script_management.py:144]: Control or automation? Consistently use the same wording. 

May be Control automation

File: python/fledge/services/core/api/filters.py [hunk 8]
```diff
@@ -267,18 +270,20 @@ async def add_filters_pipeline(request: web.Request) -> web.Response:
...
+    except ValueError as err:
+        msg = str(err)
+        raise web.HTTPNotFound(reason=msg)
+    except TypeError as err:
+        msg = str(err)
+        raise web.HTTPBadRequest(reason=msg)
     except StorageServerError as ex:
-        _LOGGER.exception("Add filters pipeline, caught exception: %s", str(ex.error))
-        raise web.HTTPInternalServerError(reason=str(ex.error))
+        msg = ex.error
```
Comment (praveen-garg) [inline python/fledge/services/core/api/filters.py:280]: may be ex should be named e.

File: python/fledge/services/core/api/filters.py [hunk 9]
```diff
@@ -315,14 +320,17 @@ async def get_filter(request: web.Request) -> web.Response:
...
         filter_detail.update({"users": users})
     except StorageServerError as ex:
-        _LOGGER.exception("Get filter: %s, caught exception: %s", filter_name, str(ex.error))
-        raise web.HTTPInternalServerError(reason=str(ex.error))
-    except ValueError as ex:
-        raise web.HTTPNotFound(reason=ex)
-    except TypeError as ex:
-        raise web.HTTPBadRequest(reason=ex)
+        msg = ex.error
+        _LOGGER.exception("Get {} filter, caught storage exception: {}".format(filter_name, msg))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/filters.py:324]: Failed to get `filter_name` filter, storage error occurred. {}

File: python/fledge/services/core/api/filters.py [hunk 10]
```diff
@@ -454,8 +470,8 @@ async def delete_filter_pipeline(request: web.Request) -> web.Response:
                 status_code = resp.status
                 jdoc = await resp.text()
                 if status_code not in range(200, 209):
-                    _LOGGER.error("Error code: %d, reason: %s, details: %s, url: %s", resp.status, resp.reason, jdoc,
-                                  put_url)
+                    _LOGGER.error("Delete {} filter pipeline; Error code: {}, reason: {}, details: {}, url: {}"
```
Comment (praveen-garg) [inline python/fledge/services/core/api/filters.py:473]: Faile to delete filters pipeline for `<service/task>`. Storage error code: ......

File: python/fledge/services/core/api/health.py [hunk 11]
```diff
@@ -39,7 +38,7 @@ async def get_disk_usage(given_dir):
     stdout, stderr = await disk_check_process.communicate()
     if disk_check_process.returncode != 0:
         stderr = stderr.decode("utf-8")
-        msg = "Failed to get disk stats! {}".format(str(stderr))
+        msg = "Failed to get disk stats of {} directory due to {}".format(given_dir, str(stderr))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/health.py:41]: "Failed to get disk stats of {} directory! {}"

File: python/fledge/services/core/api/health.py [hunk 12]
```diff
@@ -109,7 +108,7 @@ async def get_logging_health(request: web.Request) -> web.Response:
         response["levels"] = log_levels
 
     except Exception as ex:
-        msg = "Could not fetch service information.{}".format(str(ex))
+        msg = "Could not fetch service information due to {}".format(str(ex))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/health.py:111]: it was okay as is.

File: python/fledge/services/core/api/health.py [hunk 13]
```diff
@@ -208,7 +207,7 @@ async def get_storage_health(request: web.Request) -> web.Response:
         response['disk']['available'] = available
         response['disk']['status'] = status
     except Exception as ex:
-        msg = "Failed to get disk stats! {}".format(str(ex))
+        msg = "Failed to get disk stats for storage service due to {}".format(str(ex))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/health.py:210]: due to can be removed. Simply `Failed to <action e.g. get disk stats>  for <if specific>!` Followed by str(ex)

File: python/fledge/services/core/api/north.py [hunk 14]
```diff
@@ -186,6 +185,7 @@ async def get_north_schedules(request):
         return web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
     except Exception as ex:
         msg = str(ex)
+        _logger.error("Get north schedules failed. Found error: {}".format(msg))
         return web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/north.py:189]: return or raise?

File: python/fledge/services/core/api/plugins/discovery.py [hunk 15]
```diff
@@ -83,6 +84,8 @@ async def get_plugins_available(request: web.Request) -> web.Response:
         msg = "Fetch available plugins package request failed"
```
Comment (praveen-garg) [inline python/fledge/services/core/api/plugins/discovery.py:84]: is this a special case of logging warning? or it is being logged by internal interface

File: python/fledge/services/core/api/service.py [hunk 16]
```diff
@@ -365,10 +364,9 @@ async def add_service(request):
                     _logger.exception("{} Detailed error logs are: {}".format(msg, str(ex)))
                     raise web.HTTPNotFound(reason=msg, body=json.dumps({"message": msg}))
             except TypeError as ex:
-                _logger.exception(str(ex))
                 raise web.HTTPBadRequest(reason=str(ex))
             except Exception as ex:
-                _logger.exception("Failed to fetch plugin configuration. %s", str(ex))
+                _logger.error("Failed to fetch plugin configuration. %s", str(ex))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/service.py:369]: May be better ` fetch {pluginName} plugin`

File: python/fledge/services/core/api/service.py [hunk 17]
```diff
@@ -475,7 +473,7 @@ async def add_service(request):
 
             except Exception as ex:
                 await config_mgr.delete_category_and_children_recursively(name)
-                _logger.exception("Failed to create plugin configuration. %s", str(ex))
+                _logger.error("Failed to create plugin configuration. %s", str(ex))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/service.py:476]: service configuration for plugin {pluginName}

File: python/fledge/services/core/api/task.py [hunk 18]
```diff
@@ -324,9 +320,10 @@ async def delete_task(request):
         await delete_plugin_data(storage, north_instance)
         # update deprecated timestamp in asset_tracker
         await update_deprecated_ts_in_asset_tracker(storage, north_instance)
-
     except Exception as ex:
-        raise web.HTTPInternalServerError(reason=ex)
+        msg = str(ex)
+        _logger.error("Delete task failed. Found error: {}".format(msg))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/task.py:325]: Failed to delete task {task}

**Review by praveen-garg — APPROVED** [review]
Few comment to handle and mainly message construction format.

File: python/fledge/services/core/api/auth.py [hunk 19]
```diff
@@ -526,7 +513,7 @@ async def update_user(request):
 
     user_id = request.match_info.get('user_id')
     if int(user_id) == 1:
-        msg = "Restricted for Super Admin user"
+        msg = "Restricted for Super Admin user."
         _logger.warning(msg)
         raise web.HTTPNotAcceptable(reason=msg, body=json.dumps({"message": msg}))
```
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply python/fledge/services/core/api/auth.py]: >BTW, a user should be able to update her own real/display name.

They can! We have two different routes one for admin specific & for user specific

https://github.com/fledge-iot/fledge/blob/be6de2b190499dae15786b636662880f32d439ba/python/fledge/services/core/api/auth.py#L456

File: python/fledge/services/core/api/auth.py [hunk 20]
```diff
@@ -815,19 +790,18 @@ async def delete_user(request):
...
         raise web.HTTPNotFound(reason=msg)
     except Exception as exc:
-        _logger.exception(str(exc))
-        raise web.HTTPInternalServerError(reason=str(exc))
+        msg = str(exc)
+        _logger.error("Delete user failed. Found error: {}".format(msg))
+        raise web.HTTPInternalServerError(reason=msg)
 
-    _logger.info("User with id:<{}> has been deleted successfully.".format(int(user_id)))
+    _logger.info("User with ID:<{}> has been deleted successfully.".format(int(user_id)))
```
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply python/fledge/services/core/api/auth.py:800]: Yes, we need to first add new audit log entry code to use here.

File: python/fledge/services/core/api/north.py (hunk 14 above)
Comment (ashish-jabble) [reply python/fledge/services/core/api/north.py]: As there are inner try catch is being used. So we need to return early!

File: python/fledge/services/core/api/plugins/discovery.py (hunk 15 above)
Comment (ashish-jabble) [reply python/fledge/services/core/api/plugins/discovery.py]: This is the case when we have RunTimeError; especially when APT returns non zero exit code

**Pull request merged**